### ISSUE TITLE
Administrative (log-based) Statistics

### ISSUE THREAD

**Issue by heathergreerklein** [DSpace/DSpace#2852]
From release plan spreadsheet

Estimate from release plan: TBD
Expressing interest: none

Old, legacy statistics which analyzed DSpace log files using commandline scripts: https://wiki.duraspace.org/display/DSDOC6x/Command+Line+Operations#CommandLineOperations-Legacystatistics

Related UI ticket: https://github.com/DSpace/dspace-angular/issues/786

**Talk to DCAT or others about priority?**

**Comment by tdonohue** [comment]
Related to #6807

**Comment by mwoodiupui** [comment]
I'd like to observe that "statistics" is a very broad concept, with many different sources and many different consumers.  In particular, if you look at the suggested replacements above, one (Usage Statistics) is a time series and used to analyze events occurring over time which are recorded solely for statistical use, while the other (QC Reports) is a snapshot of objects currently held for other reasons and used to characterize the content of the repository at a moment in time.  I feel that, of the above suggestions, the proper choice is "both," each sticking to its own role.

There are many different uses for statistics beyond bulky reports.  Statistical measures are often used in an exploratory fashion to first figure out what should receive closer attention.  Cumulative statistics can be desirable at a variety of granularities, down to usage counts of individual objects.  So we should be careful of "premature abstraction" -- we don't know all of the uses that sites will find for such information, so we should not foreclose creative uses by making too many assumptions.  Reports, yes, but not *only* reports, and not only formatted products for humans.  (Machine-readable QC snapshot data could be collected externally and used as a time series for external analysis of trends in collection development, for example.)

I have one specific example:  a number of sites like to say "we archive upward of X thousand items" on the home page.  If that simple count (which can be done quickly and efficiently in a single database query) were available via REST, it could be inserted into the page by a simple UI component and those sites would not be forever updating their news blocks just to change a number.

All that to say this:  we need to consider how to make useful statistical products that can meet a variety of needs yet don't become the "tail wagging the dog."  It's easy to build way too much statistical analysis into a product, making creative uses difficult and overshadowing the original purpose of the product.  We should look at DSpace's own use of statistics as two components:  data gathering and presentation.  Once the data are gathered, DSpace is only one of their potential users.  This deserves new design thinking.  Simply replacing the old reports is not enough.

**Comment by tdonohue** [comment]
@mwoodiupui : You are quite correct, but I'd also want to warn us about expanding the scope of this specific ticket.

The goal of my [prior message](https://github.com/DSpace/DSpace/issues/2852#issuecomment-1286064086) was to simply note that the *3* statistical concepts in 6.x _might be_ re-envisioned as *2* concepts in 7.x.  That simplification would require first analyzing whether to use cases of 6.x log-based Statistics _can be met_ by either/both Usage Statistics & QC Reports.

Presuming that we can realign on just Usage Statistics & QC Reports, then the logical next step (after that realignment) would be to begin to determine what **additional numbers/reports** are useful.  That would open up the discussion to what you are talking about...gathering additional use cases / data needs, allowing for data export (of machine readable data), etc.

I just want to clarify that the scope of this ticket is simply to decide how to implement 6.x "log-based statistics" in the 7.x releases. The proposal from the Dev Team is that we may want to _drop_ 6.x "log-based statistics" and replace it with enhancements to "Usage Statistics" and "Quality Control Reports" which meet those same use cases.  Additional use cases (any not met by 6.x log-based statistics) would need to be described & planned in separate tickets, so that we can figure out how best to prioritize them into 7.x (or perhaps even 8.x, depending on the level of effort).

**Comment by tdonohue** [comment]
The priority of this feature was discussed in [today's DCAT meeting](https://wiki.lyrasis.org/display/cmtygp/DCAT+Meeting+November+2022).  General consensus was that this log-based statistics feature **should not be added into 7.x**.  Instead, DCAT members agreed that any of its prior use cases should be instead implemented as part of [Solr Usage Statistics](https://wiki.lyrasis.org/display/DSDOC7x/SOLR+Statistics), or [REST-based Quality Control Reports](https://wiki.lyrasis.org/display/DSDOC7x/REST+Based+Quality+Control+Reports).

No specific use cases unique to Log-based Statistics were mentioned.  However, there were a few needs mentioned regarding Statistics in general, these include:
* Desire to keep the ["DSpace Log Converter" for Solr Statistics](https://wiki.lyrasis.org/display/DSDOC7x/SOLR+Statistics+Maintenance#SOLRStatisticsMaintenance-DSpaceLogConverter).  This tool is useful for loading log file data into Solr Statistics, and might provide a "migration" path for anyone still using older log-based statistics.
* Community/Collection Statistics for 7.x should be enhanced to include aggregated statistics.  Currently, they only show statistics for the Community/Collection homepage itself.  DCAT members agreed it would be useful to show aggregated statistics for all Items in that Community/Collection (e.g. total bitstream downloads, most downloaded bitstreams, total Item accesses, most accessed Items, etc)
* DSpace should provide a way to export usage statistics into CSV files to support external analysis.  We already do provide basic export tools (e.g. ./dspace stats-util -export), which might be reasonable to use or enhance if we can provide documentation of the format of the CSV file.  https://wiki.lyrasis.org/display/DSDOC7x/SOLR+Statistics+Maintenance#SOLRStatisticsMaintenance-ExportSOLRrecordstointermediateformatforimportintoanothertool/instance

I'll create tickets for the last two feature requests.
